### PR TITLE
New ontology publication - HALO

### PR DESCRIPTION
**Pull request by navapatn**
This pull is requested to test the first version of HALO ontology

### CODE REVIEW

**Comment by dgarijo** [comment]
Please add a small description in your readme file including who the maintainer for this w3id will be.

**Comment by dgarijo** [comment]
Ping @navapatn

**Comment by navapatn** [comment]
> Ping @navapatn
 
The description and contact information were added to the readme file

File: halo-ontology/.htaccess [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+RewriteEngine On
+
+# Base directory
+RewriteBase /halo-ontology/
+
+# Rewrite rule to serve HTML content from the vocabulary URI if requested
+RewriteCond %{HTTP_ACCEPT} text/html [OR]
+RewriteCond %{HTTP_ACCEPT} application/xhtml\+xml [OR]
+RewriteCond %{HTTP_USER_AGENT} ^Mozilla/.*
+RewriteRule ^$ https://github.com/navapatn/halo-ontology/blob/main/ontology/index-en.html [R=303,L]
```
Comment (dgarijo) [inline halo-ontology/.htaccess:18]: this rewrite rule will serve a page in github. You can serve it with github pages easily, which is the target doc you are aiming for.

File: halo-ontology/.htaccess [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Rewrite rule to serve RDF/XML content from the vocabulary URI if requested
+RewriteCond %{HTTP_ACCEPT} application/rdf\+xml
+RewriteRule ^$ https://github.com/navapatn/halo-ontology/blob/main/ontology/ontology.owl [R=303,L]
```
Comment (dgarijo) [inline halo-ontology/.htaccess:22]: same issue as above

File: halo-ontology/.htaccess [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# Default response
+# ---------------------------
+# Rewrite rule to serve the RDF/XML content from the vocabulary URI by default
+RewriteRule ^$ https://raw.githubusercontent.com/navapatn/halo-ontology/main/ontology/ontology.owl [R=303,L]
+
+# Error document in case of a 406
+ErrorDocument 406 /halo-ontology/406.html
```
Comment (dgarijo) [inline halo-ontology/.htaccess:30]: /halo-ontology/406.html is not a local page. You should redirect to an html document, as you do above

**Comment by dgarijo** [comment]
@navapatn please have a look at my comments. As it is, I don't think the redirection is working as you intend

File: halo-ontology/readme.md [hunk 4]
```diff
@@ -1 +1,23 @@
-README.md draft
+HALO is built to serve as a standard stucture and model for hallucinations in generative AI. We stored ontology and entities related to hallucination in modern generative models.
+
+Our ontology can help us answer include: are LLMs relatively uniform in
+which prompts they hallucinate on, or are there significant differences between
+LLMs? What kinds of hallucination categories are most common? Furthermore,
+given longitudinal data, can we determine if LLMs are ‘improving’ on some
+categories of hallucinations, compared to others?
```
Comment (TallTed) [inline halo-ontology/readme.md:7]:
```suggestion
Questions our ontology can help us answer include: 
* Are LLMs relatively uniform in which prompts they hallucinate on, or are there 
  significant differences between LLMs?
* What kinds of hallucination categories are most common?
* Furthermore, given longitudinal data, can we determine if LLMs are ‘improving’ 
  on some categories of hallucinations, compared to others?
```

File: halo-ontology/readme.md [hunk 5]
```diff
@@ -1 +1,23 @@
...
+
+HALO is structured into two principal modules: **(1) the Hallucination module**, and **(2) the Metadata module**
+
+**The Metadata module** represents the core concept and properties of the prompts and answers we gathered such as Generative AI models, LLMs, Prompts_id, Answer_id, etc.
```
Comment (TallTed) [inline halo-ontology/readme.md:11]:
```suggestion
**The Metadata module** represents the core concepts and properties of the prompts and answers we gathered such as Generative AI models, LLMs, Prompts_id, Answer_id, etc.
```

File: halo-ontology/readme.md [hunk 6]
```diff
@@ -1 +1,23 @@
...
+
+**The Hallucination module** contains classes for each of the hallucination categories and subcategories defined as follows:
+
+1. Factuality Hallucinations refers to errors in LLMs outputs by misrepresenting real-world facts. These errors manifest in two distinct ways: Factual Inconsistency and Factual Fabrication
+2. Faithfulness Hallucination refers to LLM outputs that fail to align
+faithfully with user instructions, when provided with context, including wellknown logical rules based in objective fields like math or physics, for example.
+We further sub-categorize it into three subtypes: Instruction Inconsistency, Context Inconsistency, and Logical Inconsistency
```
Comment (TallTed) [inline halo-ontology/readme.md:18]:
```suggestion
1. _Factuality Hallucinations_ refers to errors in LLM outputs that misrepresent
real-world facts. These errors manifest in two distinct ways: _Factual
Inconsistency_ and _Factual Fabrication_.
2. _Faithfulness Hallucination_ refers to LLM outputs that fail to align faithfully
with user instructions, when provided with context, including well-known logical
rules based in objective fields like math or physics, for example. We further
sub-categorize it into three subtypes: _Instruction Inconsistency_, _Context
Inconsistency_, and _Logical Inconsistency_.
```

File: halo-ontology/readme.md [hunk 7]
```diff
@@ -1 +1,23 @@
...
+
+
+Link to our paper: https://arxiv.org/abs/2312.05209
+
+Contact information: Navapat Nananukul (nananuku@isi.edu)
```
Comment (TallTed) [inline halo-ontology/readme.md:23]:
```suggestion
Contact information: Navapat Nananukul (nananuku@isi.edu) [@navapatn](https://github.com/navapatn) 
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Some tweaks for humans...

**Comment by navapatn** [comment]
@TallTed @dgarijo 
Thank you for both of your suggestions. 
I have made all the changes both in .htaccess and readme.md

File: halo-ontology/.htaccess [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+RewriteCond %{HTTP_ACCEPT} application/rdf\+xml
+RewriteRule ^$ https://navapatn.github.io/halo-ontology/ontology/ontology.owl [R=303,L]
+
+# Default response
+# ---------------------------
+# Rewrite rule to serve the RDF/XML content from the vocabulary URI by default
+RewriteRule ^$ https://raw.githubusercontent.com/navapatn/halo-ontology/main/ontology/ontology.owl [R=303,L]
+
+# Error document in case of a 406
+ErrorDocument 406 https://navapatn.github.io/halo-ontology/ontology/406.html
```
Comment (dgarijo) [inline halo-ontology/.htaccess:30]: Have you validated this rule? I do not think line 30 is valid. See https://github.com/perma-id/w3id.org/blob/master/example/.htaccess for an example

**Comment by dgarijo** [comment]
@navapatn 
Please see my comment about line 30 in your htaccess file

**Comment by navapatn** [comment]
@dgarijo Thank you for the example, and I'm sorry for the confusion. I have adjusted the file based on the example.

**Pull request merged**